src/realm/views.cpp

```diff
@@ -319,36 +350,85 @@ void DescriptorOrdering::append_distinct(DistinctDescriptor distinct)
     }
 }
 
+void DescriptorOrdering::append_limit(LimitDescriptor limit)
+{
+    REALM_ASSERT(limit.is_valid());
+    m_descriptors.emplace_back(new LimitDescriptor(std::move(limit)));
+}
+
 bool DescriptorOrdering::descriptor_is_sort(size_t index) const
 {
     REALM_ASSERT(index < m_descriptors.size());
-    SortDescriptor* sort_descr = dynamic_cast<SortDescriptor*>(m_descriptors[index].get());
-    return (sort_descr != nullptr);
+    auto& temp = *m_descriptors[index].get(); // a temporary to workaround a clang warning
+    return typeid(temp) == typeid(SortDescriptor);
```

Comment:
Perhaps a bit old-fashioned but easier to read if you would call a virtual function returning DescriptorType and then compare with DescriptorType::Sort. This comment applies in many places below as well.